### PR TITLE
Moved "How To" and other useful documentation in-app

### LINKED ISSUE #874: Move How-to Docs about NBG directly with the app
## Observed
1. All of our guides to do things in the base app are covered in our supplemental hidden documentation but it seems that other people using the app might benefit from having access to this.
2. In addition it's been observed that our own folks are also having issues finding the documentation.

## Proposal
1. Move all videos, images, docs, guides, etc. to live under `app/views/docs/*` in codebase to live under ex. `https://nbgallery.com/docs/BLAH` URL for the user. 
2. Link to this from the question mark dropdown in header.
3. Add config to disable it for site admins that do not want it.
4. (MAYBE) Allow users to switch from searching notebooks to searching in the docs seamlessly.
5. Make all docs able to be accessed by anonymous users but hide from search engines... I think.

## Benefits
1. Will be much easier to maintain between versions because as soon as new version hits with new functionality, the docs would be updated too.
2. All partners and people making their own NBGallerys would have access to the docs as well without any access problems (because it lives directly within the app).
3. ENSURES documentation that goes with the app is accessible. If docs are in multiple formats, PDF, word, etc. or on a portal or KB system that isn't fully accessible, it doesn't matter how accessible the docs are. If our system is accessible and our docs are too, we can ensure we're making it as easy as possible for persons with disabilities.

### PR DESCRIPTION
**Pull request by eldenlord204824**
I was unable to really find the documentation referenced in #874. Therefore, as an alternative I have made simple overarching documentation with the ability for enterprises to add their own documentation appended to the end of the initial documentation.

I am open to feedback, proof reading, corrections, and additional overarching documentation that may need to be added or an alternative way to approach this issue.

Note: In order to view review documentation remove the conditional in /app/views/docs/notebook_review.slim and use endpoint /docs/notebook_review.

Will close #874

### CODE REVIEW

File: app/views/docs/utilizing_notebooks.slim [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  div.content-container id="uploadingNotebook"
+    h3 Uploading a Notebook
+    p Uploading a noteboook is easy and allows the use of tags to help users find the notebooks that will help the most.
+    ol
+      li Press the upload button on top right of the screen, and upload your notebook from your desktop.
+      li From here, press next and various fields will populate. 
+      ul
+        li <strong>Notebook ID:</strong> the unique Notebook Identifier (this cannot be altered/changed)
+        li <strong>Title:</strong> the title of the notebook. (this can be different from file name)
+        li <strong>Desciption:</strong> each notebook is required to have a description to help user understand the use of the notebook and the particular how tos of the specific notebook.
```
Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: Description*

File: app/views/docs/utilizing_notebooks.slim [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  h1.text-center Utilizing Notebooks
+  p 
+    | The main purpose of NBGallery is to provide a repository of notebooks to be utilized by users for various purposes in 
+    | the data science community. The basic features and how tos are essiential to use NBGallery to it's full potential. 
+    | Viewing the notebooks in NBGallery is as simple as hitting the button labeled notebooks, and to find more nuanced selections 
+    | (user notebooks, all groups, all laguages, all tags, and your groups) hit the dropdown arrow to the right of the button. 
+  br/
+  div.content-container id="uploadingNotebook"
+    h3 Uploading a Notebook
+    p Uploading a noteboook is easy and allows the use of tags to help users find the notebooks that will help the most.
```
Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim:11]: notebook*

File: app/views/docs/utilizing_notebooks.slim [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        li <strong>Title:</strong> the title of the notebook. (this can be different from file name)
+        li <strong>Desciption:</strong> each notebook is required to have a description to help user understand the use of the notebook and the particular how tos of the specific notebook.
+        li <strong>Ownership:</strong> ownership by default is granted to the uploader, but can also be granted to a certain group and another user entirely (this can be changed later if needed).
+        li <strong>Tags:</strong> tags, while not required, do help users, at a quick glance, add context to what a notebook contains and decide if your notebook may be of use to them.
+        li <strong>Visibility:</strong> you can make your notebook visible to only yourself or the public.
+      li Once all required fields are filled in press upload, and it will lead you directly to your notebook on page and success notification.
+  br/
+  div.content-container id=:"notebookPage"
+    h3 The Notebook Page
+    p Everything need to know about any particular notebook can be found on the page of the notebook itself.
```
Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: > "Everything _**you**_ need to know ... "

File: app/views/docs/utilizing_notebooks.slim [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    p There are options and features for each notebook as an owner/group member of notebook and a casual user view.
+    p The various items are pretty self explainitory, including basic notbook information such as the title, owner (and creator if applicable) description, tags, main language of the notebook, star and view count, and current health status of the notebook.
```
Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: self-explanatory*

Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: notebook*

File: app/views/docs/utilizing_notebooks.slim [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    p The tool bar seen to the right side under the "Run in Jupyter" button allows for useful functions.
+    ul
+      li <strong>Star:</strong> star is the equivalent of a like on NBGallery. you are able to view an notebooks you have starred as an NBGallery user.
+      li <strong>Subscribe:</strong> subcribing to a particular notebook gives you the option to recieve update notification on particular notebooks (maybe ones you utilize a lot)
```
Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: subscribing*

Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: receive*

File: app/views/docs/utilizing_notebooks.slim [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      li <strong>Download:</strong> download the notebook using this option.
+      li <strong>Feedback:</strong> allows you to send direct feedback of improvments that could be made or well done elements of a notebook to the owner or group.
+      li <strong>Comments:</strong> allows you to comment on a particular notebook (a question, general advise, or just a compliment if you like).
```
Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: advice*

File: app/views/docs/utilizing_notebooks.slim [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      li <strong>Settings:</strong> dropdown includes more options for a particular notebook (some only available, if you own the notebook or not).
+      ul
+        li <strong>View Notebook Metrics:</strong> this leads you to an abundance of notebook information about the notebook including runs, user views (and total views), edit history, number of stars, and other metrics.
+        li <strong>View Revision History:</strong> this leades you to the changes to notebook and notices about changes (creation, deprecation, and general edits along with who made did it)
```
Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: leads*

Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim]: > "... and general edits along with who made **_the changes_**)"

File: app/views/docs/utilizing_notebooks.slim [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          li <strong>Make private:</strong> make a notebook private
+          li <strong>Change ownership:</strong> change the ownership to another user or group
+          li <strong>Share notebook with individuals:</strong> share your notebook with individuals (cannot share private will not be viewable)
+          li <strong>Upload new version:</strong> update the contents of the notebook by uploading the latest version.
+          li <strong>Delete notebook:</strong> delete a notebook (this action cannot be undone!)
+          li <strong>Deprecate Notebook:</strong> deprecate a notebook. this option is similar to deleting except the notebook will stay a part of NBGallery (you can chose to allow users to run the notebook or not). the contents of a deprecated notebook may be outdated or have other deprecated elements (this deprecation can be reversed)
+  div.content-container id="searchingNotebooks"
+    h3 Searching Notebooks
+    p What good is having a collection of notebooks if we cannot filter through them to find only the most critical/useful one. Searching is made easy and seemless in NBGallery.
```
Comment (mild-bit) [inline app/views/docs/utilizing_notebooks.slim:54]: seamless*

**Review by mild-bit — CHANGES_REQUESTED** [review]
fix typos and grammatical errors

**Review by ace6150 — APPROVED** [review]

**Review by mild-bit — APPROVED** [review]

**Pull request merged**